frame/staking/src/lib.rs

```diff
@@ -2870,249 +2254,46 @@ impl<T: Config> Module<T> {
 		}
 
 		// Set staking information for new era.
-		let maybe_new_validators = Self::select_and_update_validators(current_era);
-		// TWO_PHASE_NOTE: use this later on.
-		let _unused_new_validators = Self::enact_election(current_era);
+		let maybe_new_validators = Self::enact_election(current_era);
 
 		maybe_new_validators
 	}
 
-	/// Remove all the storage items associated with the election.
-	fn close_election_window() {
-		// Close window.
-		<EraElectionStatus<T>>::put(ElectionStatus::Closed);
-		// Kill snapshots.
-		Self::kill_stakers_snapshot();
-		// Don't track final session.
-		IsCurrentSessionFinal::put(false);
-	}
-
-	/// Select the new validator set at the end of the era.
-	///
-	/// Runs [`try_do_phragmen`] and updates the following storage items:
-	/// - [`EraElectionStatus`]: with `None`.
-	/// - [`ErasStakers`]: with the new staker set.
-	/// - [`ErasStakersClipped`].
-	/// - [`ErasValidatorPrefs`].
-	/// - [`ErasTotalStake`]: with the new total stake.
-	/// - [`SnapshotValidators`] and [`SnapshotNominators`] are both removed.
-	///
-	/// Internally, [`QueuedElected`], snapshots and [`QueuedScore`] are also consumed.
-	///
-	/// If the election has been successful, It passes the new set upwards.
-	///
-	/// This should only be called at the end of an era.
-	fn select_and_update_validators(current_era: EraIndex) -> Option<Vec<T::AccountId>> {
-		if let Some(ElectionResult::<T::AccountId, BalanceOf<T>> {
-			elected_stashes,
-			exposures,
-			compute,
-		}) = Self::try_do_election() {
-			// Totally close the election round and data.
-			Self::close_election_window();
-
-			// Populate Stakers and write slot stake.
-			let mut total_stake: BalanceOf<T> = Zero::zero();
```

Comment:
@thiolliere I have re-intorduced this piece of logic in `process_election`, but this is overall one of my main worries about the refactor.